lib/niconico/video.rb

```diff
@@ -86,7 +86,18 @@ def available?
 
     def get_video
       raise VideoUnavailableError unless available?
-      @agent.get_file(video_url)
+      unless block_given?
+          @agent.get_file(video_url)
+      else
+        cookies = @agent.cookie_jar.cookies('http://nicovideo.jp').map{|c|c.to_s}.join(';')
```

Comment:
- `@agent.cookie_jar.cookies('http://nicovideo.jp')` can be replaced with `video_cookies` (defined below)
- `map(&:to_s)`